### PR TITLE
Add output data streams to python functions

### PR DESCRIPTION
**Pull request by SvenO3**
### Purpose

This PR implements the support for output data streams for the python functions. This makes it possible to create a new data stream out of python and to write back live data to StreamPipes via that stream.
It also provides an implementation to use Online Machine Learning with the StreamPipesFunctions using the River library. You can now train an Online ML model on a data stream and sends the prediction back to StreamPipes with very little effort.

 ### Remarks

PR introduces (a) breaking change(s): no

PR introduces (a) deprecation(s): no

### CODE REVIEW

File: streampipes-client-python/streampipes_client/functions/machine_learning/__init__.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
```
Comment (bossenti) [inline streampipes-client-python/streampipes_client/functions/machine_learning/__init__.py:2]: lets move this into a dedicated module called `function_zoo`

**Review by bossenti — COMMENTED** [review]
great work @SvenO3  🥇

**Comment by bossenti** [comment]
almost forgot: please rebase the PR before we merge it because we upgraded `black`

File: streampipes-client-python/streampipes_client/functions/machine_learning/__init__.py (hunk 1 above)
Comment (SvenO3) [reply streampipes-client-python/streampipes_client/functions/machine_learning/__init__.py:2]: Should `function_zoo` be a submodule of the `functions` module. Or should it be seperated to differentiate between implementation and examples?

Comment (bossenti) [reply streampipes-client-python/streampipes_client/functions/machine_learning/__init__.py:2]: I would see it as separate module on the same level like functions.
Should also be beneficial in terms of imports

**Comment by bossenti** [comment]
Needs to be merged in sync with https://github.com/apache/streampipes/pull/1228

**Comment by SvenO3** [comment]
> Needs to be merged in sync with #1228

What's the best way to do this?

**Comment by bossenti** [comment]
Well, one of us has to do some work 😅
Either we merge #1228 first, then you need to update this PR accordingly or we do it the way around :)

**Pull request merged**